Renew PrestaShop back-office branding

tblivet opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project!

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/#pull-requests

For type and category see:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#type--category
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Renew PrestaShop back-office branding, agreed by @matks. See https://build.prestashop-project.org/news/2023/new-project-branding/ and https://github.com/PrestaShop/PrestaShop/issues/36615
| Type?             | new feature
| Category?         | BO
| BC breaks?        | 
| Deprecations?     | 
| How to test?      | All the back office is rebranded base on the new brand identity of PrestaShop
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/10270851789
| Fixed issue or discussion?     | 
| Related PRs       | 
| Sponsor company   | @PrestaShopCorp

tblivet committed:
feat: rebrand themes

tblivet committed:
feat: images update

tblivet committed:
feat: PrestaShopBundle admin rebrand

tblivet committed:
feat: PrestaShopBundle admin form rebrand

tblivet committed:
feat: custom specific rebrand

tblivet committed:
feat: review WIP

tblivet committed:
feat: add PrestaShop version CSS class

ps-jarvis commented:
<!-- PR_TABLE_DESCRIPTION_ERROR -->
Hi, thanks for this contribution!

I found some issues with the Pull Request description:

* The `BC breaks` should be `yes` or `no`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#bc-breaks))
* The `deprecations` should be `yes` or `no`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#deprecations))

Would you mind having a look at it? This will help us understand how interesting your contribution is, thank you very much!
### About linked issues
Please consider opening an issue before submitting a Pull Request:
* If it's a bug fix, it helps maintainers verify that the bug is effectively due to a defect in the code, and that it hasn't been fixed already.
* It can help trigger a discussion about the best implementation path before a single line of code is written.
* It may lead the Core Product team to mark that issue as a priority, further attracting the maintainers' attention.

(Note: this is an automated message, but answering it will reach a real human)

jf-viguier commented:
It's a huge change that could have impact on bo module developers so this PR should be more documented and explained.

matks commented:
> It's a huge change that could have impact on bo module developers so this PR should be more documented and explained.

Hello @jf-viguier yes you are right, this will have an impact of the backoffice.

@tblivet will document this change in the Developer Documentation 😉 . This will allow developer willing to learn "what's changed between PrestaShop 8 and PrestaShop 9" to be informed and helped.

kpodemski commented on admin-dev/themes/default/js/admin-theme.js:
```diff
@@ -510,4 +510,28 @@ $(() => {
+  updatePadding();
+
+  // Update padding when the window is resized
+  window.addEventListener('resize', updatePadding);
```
I wonder if it wouldn't be a good idea to use debounce on this one

tblivet replied:
Debounce has been added ✅

kpodemski commented on admin-dev/themes/default/scss/partials/_commons.scss:
```diff
@@ -1,312 +1,24 @@
-// helpers class for fixing input width
```
For others who review, they are not deleted, they went to other file 😅

kpodemski commented on admin-dev/themes/default/scss/partials/_forms.scss:
```diff
@@ -47,6 +46,33 @@ textarea {
+input[disabled][type="text"],
+input[readonly][type="password"],
+input[readonly][type="search"],
+input[readonly][type="text"],
```
Maybe you should also add [type="email"], [type="number"]?

tblivet replied:
[type="email"], [type="number"] has been added ✅

kpodemski commented on admin-dev/themes/default/template/header.tpl:
```diff
@@ -262,9 +262,7 @@
              title="{$smarty.capture.title|htmlspecialchars}"
           >
-            <i class="material-icons"
-               style="{if isset($maintenance_allow_admins)}color: #72c279;{/if}"
```
As I mentioned in the pre-review, this should be highlighted one way or another. It should stay because it's a feature added recently that allows admins to access the store even if the maintenance is enabled and their IP hasn't been added to the allow list.

tblivet replied:
I have highlighted the maintenance link in the header in CSS after a talk with designers ✅
However, I think we have an issue because the maintenance link in the header seems to never be displayed (with or without the rebranding).

jolelievre replied:
Yep and thanks for opening an issue about that
But regardless, previously the color changed depending on the `maintenance_allow_admins` value, since you removed the condition here it means the color is always the same regardless of this variable right? Which is a modification of the initial behaviour

tblivet replied:
Yes, you're right, but according to designers we will use only one color for this.

kpodemski commented on admin-dev/themes/default/template/nav.tpl:
```diff
@@ -1,7 +1,6 @@
 	<span class="menu-collapse" data-toggle-url="{$toggle_navigation_url}">
-		<i class="material-icons rtl-flip">chevron_left</i>
-		<i class="material-icons rtl-flip">chevron_left</i>
+    <i class="material-icons rtl-flip">first_page</i>
```
Something went wrong with indentation.

tblivet replied:
✅

Hlavtox reviewed (changes requested):
For me, the branding is very bland and not very user friendly, as everything is just black. But there is no point in fighting this.

So, please fix styling of some elements:

**Buttons:**
- btn-warning incosistent, has a border.
- btn-dark completely disappears on hover.
- btn-outline-success and btn-outline-warning and btn-outline-info incosistent hover.
- btn-outline-dark completely disappears on hover

The original styling can be used as a reference - https://getbootstrap.com/docs/4.6/components/buttons/,

**Alerts:**
- alert-dark is not styled, looks the same as light.
https://getbootstrap.com/docs/4.6/components/alerts/

kpodemski commented on admin-dev/themes/new-theme/scss/components/_grid.scss:
```diff
@@ -322,15 +359,27 @@ table {
-// -----------------------
-// Styling grid extensions
-// -----------------------
+      td[data-column-id^="id_product"],
```
Hardcoded values like this do not look good. Can we do anything on the Grid level to help you style it with a more generic class?

tblivet replied:
I removed those IDs; it should work the same without any breaking changes ✅
However, the IDs were already used on certain elements in this file, but you're right, it's not ideal.

kpodemski commented on admin-dev/themes/new-theme/scss/components/_hint_box.scss:
```diff
@@ -1,8 +0,0 @@
-.hint-box {
```
Used inside

`src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_4_layout.html.twig`

Maybe you changed it later and I don't see it, but I'm leaving the comment anyway.

tblivet replied:
The style of this `.hint-box` is like an alert with a thicker left border, which to me makes no sense because we are trying to harmonize the BO. Therefore, it has no impact for me if the hint box displays as an alert, because the Twig file `bootstrap_4_layout.html.twig` uses this markup: `<div class="alert alert-info">`, so it will render as an info alert. That's why I have removed this part.

kpodemski commented on admin-dev/themes/new-theme/template/layout.tpl:
```diff
@@ -79,9 +79,7 @@
              href="{$link->getAdminLink('AdminMaintenance')|escape:'html':'UTF-8'}"
           >
-            <i class="material-icons"
-              style="{if isset($maintenance_allow_admins) && $maintenance_allow_admins}color: var(--green);{/if}"
```
You have to keep it.

tblivet replied:
I have highlighted the maintenance link in the header in CSS after a talk with designers ✅

kpodemski commented on src/PrestaShopBundle/Resources/views/Admin/Layout/default_layout.html.twig:
```diff
@@ -38,7 +38,16 @@
-  <body class="lang-{{ ps.isoUser }}{% if ps.isRtlLanguage %} lang-rtl{% endif %} {{ ps.controllerName|escape|lower }}{% if ps.menuCollapsed %} page-sidebar-closed{% endif %}{% if ps.multiShop|default(false) %} multishop-enabled{% endif %}{% if ps.debugMode|default(false) %} developer-mode{% endif %}"
+
+  {# Construct CSS classes based on version numbers #}
+  {% set major_version = constant('PrestaShop\\PrestaShop\\Core\\Version::MAJOR_VERSION') %}
```
I think you could ask core developer to do it cleanly on the back end side of things.

tblivet replied:
Ok, I have asked @jolelievre what he thinks about that.

jolelievre replied:
We indeed have something similar already available `{{ ps.version }}` But it contains the whole version, however careful as @kpodemski said this one shouldn't be used in the login page as we try to limit the number of elements that help defining the PrestaShop version on public pages

Inside the back office it should be fine we could add something like `{{ ps.majorVersion }}` and so on for minor and patch versions. But I would need to have more details about your need Why do you need a body class that depends on the version I didn't see any reference in the PR (I may have missed them), is it required in the UI kit?

kpodemski commented on src/PrestaShopBundle/Resources/views/Admin/Layout/default_layout.html.twig:
```diff
@@ -112,9 +121,7 @@
              href="{{ url('admin_maintenance') }}"
           >
-            <i class="material-icons"
-               style="{% if ps.frontOfficeAccessibleForAdmins|default(false) %}color: var(--green);{% endif %}"
```
You need to keep it.

tblivet replied:
I have highlighted the maintenance link in the header in CSS after a talk with designers ✅

jolelievre replied:
Same comment as in the default theme, how do you handle the dynamic value depending on `frontOfficeAccessibleForAdmins` with pure CSS?

Or did the designers decide that this feature was not relevant and decided to remove it completely?

kpodemski replied:
@jolelievre it's been removed now, IMHO designers should adapt to existing features, not remove features 😅 it's probably just a mistake and not knowing about this particular scenario available

tblivet replied:
According to designers, we will use only one color for this.

jolelievre replied:
Ok why not, indeed I'm not sure just changing a color is that meaningful anyway
@kpodemski even though I agree with you on the principal in this particular case do you think it is critical to keep this feature?

kpodemski replied:
@jolelievre, it definitely is not critical to have a different color for the link, if they don't think it is relevant for back office users to understand that the front of the store is visible only to them

jolelievre replied:
Yep I think it's ok as well

kpodemski commented on src/PrestaShopBundle/Resources/views/Admin/Layout/legacy_layout.html.twig:
```diff
@@ -34,7 +34,15 @@
 
-<body class="lang-{{ ps.isoUser }}{% if ps.isRtlLanguage %} lang-rtl{% endif %} ps_back-office page-sidebar {{ ps.controllerName|escape|lower }}{% if ps.menuCollapsed %} page-sidebar-closed{% endif %}{% if ps.multiShop|default(false) %} multishop-enabled{% endif %}{% if ps.debugMode|default(false) %} developer-mode{% endif %}"
+{# Construct CSS classes based on version numbers #}
+{% set major_version = constant('PrestaShop\\PrestaShop\\Core\\Version::MAJOR_VERSION') %}
```
Same as for the previous comment, better ask core dev to do it on the back end.

jolelievre replied:
See previous comment in default layout

kpodemski commented on src/PrestaShopBundle/Resources/views/Admin/Layout/legacy_layout.html.twig:
```diff
@@ -96,9 +104,7 @@
              href="{{ url('admin_maintenance') }}"
           >
-            <i class="material-icons"
-               style="{% if ps.frontOfficeAccessibleForAdmins|default(false) %}color: #72c279;{% endif %}"
```
You need to keep it.

tblivet replied:
I have highlighted the maintenance link in the header in CSS after a talk with designers ✅

kpodemski commented on src/PrestaShopBundle/Resources/views/Admin/Layout/light_layout.html.twig:
```diff
@@ -39,8 +39,16 @@
   {% endblock %}
 </head>
 
+{# Construct CSS classes based on version numbers #}
```
Having this code in three files proves the point of having to separate it somewhere upper 😅

kpodemski commented on src/PrestaShopBundle/Resources/views/Admin/Layout/login_layout.html.twig:
```diff
@@ -34,21 +34,34 @@
-<body class="lang-{{ ps.isoUser }}{% if ps.isRtlLanguage %} lang-rtl{% endif %} ps_back-office bootstrap">
+
+{# Construct CSS classes based on version numbers #}
+{% set major_version = constant('PrestaShop\\PrestaShop\\Core\\Version::MAJOR_VERSION') %}
```
Again.

jolelievre replied:
In this case it is more a problem, as I commented above for security reasons we try not to emphasize on the version of the shop It doesn't mean there are no other ways to determine the PrestaShop version of course, but at least it avoids making things too easy/direct

kpodemski replied:
it's ok after login to back office, it should be hidden on the login page

kpodemski commented on src/PrestaShopBundle/Resources/views/Admin/Layout/login_layout.html.twig:
```diff
@@ -34,21 +34,34 @@
-          <h1 class="text-center">
-            <img id="logo" src="{{ ps.baseImgUrl }}prestashop@2x.png" width="123px" height="24px" alt="PrestaShop" />
+          <h1 class="text-center mb-0">
+            <img id="logo" src="{{ ps.baseImgUrl }}prestashop@2x.png" width="128" height="auto" alt="PrestaShop" loading="lazy" />
```
It's above the fold, you should not lazy load this image.

tblivet replied:
Yes, you're right ✅

kpodemski commented on src/PrestaShopBundle/Resources/views/Admin/Layout/login_layout.html.twig:
```diff
@@ -34,21 +34,34 @@
+          <h1 class="text-center mb-0">
+            <img id="logo" src="{{ ps.baseImgUrl }}prestashop@2x.png" width="128" height="auto" alt="PrestaShop" loading="lazy" />
           </h1>
+          <div class="login__version text-center">
```
Login page should not include ps_version.

tblivet replied:
Ok, I have removed this part ✅

kpodemski commented on src/PrestaShopBundle/Resources/views/Admin/Layout/login_layout.html.twig:
```diff
@@ -99,15 +112,14 @@
           <span class="login-back-shop">{{ shopName }}</span>
         </a>
 
-        {{ renderhook('displayAdminLogin') }}
```
I'm not sure about this, I think this hook should be kept, right @jolelievre ?

tblivet replied:
The hook has been re-added ✅
Sorry, I think I removed it during a rebase.

kpodemski commented on src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/prestashop_ui_kit_base.html.twig:
```diff
@@ -1460,8 +1460,7 @@
         {% if child.vars.label_tab is defined %}
-          {# We force required false here so that the required asterix is never shown on the tab title #}
-          {{ form_label(child, child.vars.label_tab, { required: false }) }}
+          {{ form_label(child, child.vars.label_tab) }}
```
I am not sure about this change, `required: false` is there for a reason.

tblivet replied:
Re-added ✅

jolelievre commented on admin-dev/themes/default/example/components/alerts.html:
```diff
@@ -0,0 +1,22 @@
+<h1 class="custom">Alerts</h1>
```
Are these example files still useful?

I believe you used them during development, but now that it's done maybe they should be cleared?

tblivet replied:
I think these examples can be useful as a tool for the future if we need to edit default theme.

jolelievre commented on admin-dev/themes/default/fonts/MaterialSymbolsOutlined[FILL,GRAD,opsz,wght].ttf:
Do these asset files need to be persisted in the repository? Or could they be integrated via packages instead?

Usually the assets dependencies are not "hard persisted" when we can integrate them via a package manager

jolelievre commented on admin-dev/themes/default/package.json:
```diff
@@ -1,56 +1,59 @@
+    "@openfonts/ubuntu-condensed_latin": "^1.44.2",
+    "open-sans-fonts": "^1.6.2",
+    "perfect-scrollbar": "^1.4.0",
+    "prestakit": "tblivet/prestashop-ui-kit_fork#reskin_update",
```
Ok for now but this should be replaced with a real version once the prestashop UI kit is updated and released, e have two possible strategies here:
- merge this PR as is, wait for the UI kit PR to be merged (and released), and then do a small PR to update this requirement and update dependencies
- wait for the UI kit PR to be merged (and released), and update this dev dependencies before we merge this PR

tblivet replied:
Yes, you are right. I spoke with @matks about this. I think we will wait for the UI Kit to be merged and released as v2.0.0, and then update dependencies before the merge.

jolelievre commented on admin-dev/themes/default/postcss.config.js:
```diff
@@ -1,29 +0,0 @@
- * @copyright Since 2007 PrestaShop SA and Contributors
- * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
- */
-module.exports = {
```
Why remove this file? Is it not used anymore?

tblivet replied:
The package has been deprecated, and I don't think this is still relevant.

jolelievre commented on admin-dev/themes/default/scss/controllers/_carrier-wizard.scss:
```diff
@@ -36,7 +37,7 @@
     width: 100%;
     padding: 0;
-    margin: 0 0 10px;
+    margin: 0 0 var(--#{$cdk}size-10);
```
What's the interest of having this variable? Seems like it's named after its actual value, not its purpose What's the advantage of using a hard-coded variable which name is based on the value instead of the value itself?

The variable value can be changed but does it make sense to change the value of `--#{$cdk}size-10` to `16px` for example?

tblivet replied:
These variables are defined in `rem`. They are mainly (though not exclusively) based on multiples of 4. In the future, I think this will lead to better harmonization of spacing and size if we use these variables instead of arbitrary values like (13px / 19px or others).

jolelievre commented on admin-dev/themes/default/scss/controllers/_customer-thread.scss:
```diff
@@ -212,12 +211,12 @@ $time-icon-size: 40px;
     content: "";
-    background-color: #ddd;
-    @include margin-left(-4px);
+    background-color: var(--#{$cdk}primary-400);
```
Similarly here, what does the `primary-400` stand for?

tblivet replied:
`primary-400` refers to a variation of the new primary color, which is a shade of gray. This is a first small step towards a potential customization of the back office.

jolelievre commented on admin-dev/themes/default/scss/controllers/_login.scss:
```diff
@@ -1,21 +1,26 @@
   min-height: 100%;
-  padding-bottom: 45px;
-}
+  padding-bottom: var(--#{$cdk}size-40);
```
The login page has been migrated, it's now independent from the default theme so I wonder if these modifications are still needed?
This file should maybe be removed now (not in this PR's scope though)

tblivet replied:
Yes, I did it before the page was migrated to Symfony, which is why the code is also present in the default theme. So yes, it can be removed.

jolelievre commented on admin-dev/themes/default/scss/controllers/_modules.scss:
```diff
@@ -157,15 +157,15 @@ td.module_inactive {
 
 .module-badge-popular {
-  background-color: $brand-addons;
+  background-color: var(--#{$cdk}blue-500);
```
I'm not sure to understand the advantage of this kind of variables compared to the previous ones which had semantic meaning and context

Also now we can't change the color of modular badge independently, if I modify `blue-500` it will impact all the elements that need this blue color but it's probably gonna affect too many CSS definitions Isn't it a problem?

tblivet replied:
The idea behind these variables was to have a single source of truth, which comes from the UI Kit. However, as I mentioned in a previous comment, this is just the first step towards potential customization. If we want to go further, we will need to create design tokens and use them instead of generic variables. I'm available to discuss this with you.

jolelievre commented on admin-dev/themes/default/scss/font.scss:
```diff
@@ -1,35 +1,102 @@
+$local-font: true;
+
+@mixin setFont() {
+  @if $local-font {
```
This seems to always be hard-coded to `true`, so is the `if/else` logic relevant here?

tblivet replied:
The logic was already in place before the rebrand, so I just updated it.

jolelievre commented on admin-dev/themes/default/scss/modules/colors/_ambers.scss:
```diff
@@ -1,25 +0,0 @@
-// Ambers 9
-// -------------------------
-
-$amber:               	#f90 !default;
```
Are these variables not used anymore or were they moved somewhere else?

in both case this is still a breaking change because some module may expect this file to be present and try to import it It doesn't mean e can't remove this file (since we're in a major version), but the PR must indicate it contains some breaking changes so that we can document it, and ideally you need to explain this breaking change (explain if the variables are completely removed or if they can be imported from another file).

kpodemski replied:
nice catch @jolelievre, I haven't seen anything like this but it is possible that somewhere, someone... :)

tblivet replied:
Yes, I removed this part because I didn't see any occurrences of it across the back office. I will document the rebranding in the devdocs, where I can mention it.

jolelievre replied:
I don't mind cleaning these files indeed, but we need to document this as a breaking change

jolelievre commented on admin-dev/themes/default/scss/partials/_commons.scss:
```diff
@@ -1,312 +1,24 @@
-// helpers class for fixing input width
-.fixed-width-xs {
```
Does it mean these helper classes are no longer usable? Why remove them?
If it's relevant it should at least be documented as a breaking change

jolelievre replied:
Seems like, at least some of them, were moved into `_helpers.scss` file, so this should be indicated in the PR description so it's not forgotten and so we can update the dev doc based on this PR's description later

kpodemski replied:
is it really necessary? it shouldn't change anything for the developers

tblivet replied:
No, they are not removed, just move in a `_helpers.scss` file.
For the parts that are no longer in helpers, they are included in a better place.
For example: `.alert.alert-danger {...` as been move to `_alerts.scss` file...

jolelievre commented on admin-dev/themes/default/scss/partials/_nav.scss:
```diff
@@ -16,64 +14,78 @@
+        width: var(--#{$cdk}size-128);
+        height: var(--#{$cdk}size-24);
+        //background: url("#{$img-path}/ps-logo-white.svg") no-repeat;
+        background: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAUYAAAAkCAMAAADCQmMnAAAAAXNSR0IB2cksfwAAAAlwSFlzAAAWJQAAFiUBSVIk8AAAAEVQTFRFHR0bHR0bHR0bHR0bHR0bHR0bAAAAHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0buFvkHQAAABd0Uk5Tv6+fgFAgAEB/3+//z2AQj3BvXzCQoLC2y4BfAAAGb0lEQVR4nLWa6aKrKAyA3TngUu/oue//qFO1ZCNBOu3kXwGzfOyhVcWlbtquH4g0N/XDUBnS0Ubup/ZhHMNUz+RzZ317WIJWP1jYDJos ...
```
Is it better to integrate the image with its base64 value here? Why not use an image file instead? It would probably be easier to maintain

tblivet replied:
This was done a long time ago with the intent of simply editing the CSS file, but I've gone a bit further since then by modifying a few template, images... So, I'm going to refactor this to use a real SVG instead of base64.

tblivet committed:
fix: rebrand review

tblivet committed:
refactor: button

tblivet commented:
> For me, the branding is very bland and not very user friendly, as everything is just black. But there is no point in fighting this.
> 
> So, please fix styling of some elements:
> 
> **Buttons:**
> 
> * btn-warning incosistent, has a border.
> * btn-dark completely disappears on hover.
> * btn-outline-success and btn-outline-warning and btn-outline-info incosistent hover.
> * btn-outline-dark completely disappears on hover
> 
> The original styling can be used as a reference - https://getbootstrap.com/docs/4.6/components/buttons/,
> 
> **Alerts:**
> 
> * alert-dark is not styled, looks the same as light.
>   https://getbootstrap.com/docs/4.6/components/alerts/

Hi @Hlavtox,

The rebranding of the elements you mentioned is based on the PrestaShop UI Kit Storybook. Some of the elements were not included, so I've added the ones you mentioned and refactored the `alerts` and `buttons` components.

Furthermore, a PR has been opened here: https://github.com/PrestaShop/prestashop-ui-kit/pull/229 in case you want to run Storybook to check the rebranding changes.

jolelievre commented on admin-dev/themes/new-theme/img/empty_state/combinations.svg:
I like this visual, but to me there is a problem the price is displayed in euros
PrestaShop is an international solution, the euro won't make sense for any country outside of the EU

If anything if we have to keep a single symbol integrated in the SVG the $ has a more universal meaning overseas

tblivet replied:
Ok it makes sens, I will ask for designers what do they think.

jolelievre commented on admin-dev/themes/new-theme/img/empty_state/combinations_history.png:
The other PNGs were removed in favor of the SVGs
Any reason why this one is kept? Couldn't it be replaced by combinations_history.svg ?

tblivet replied:
Good catch, that was an oversight.

jolelievre commented on admin-dev/themes/new-theme/img/empty_state/missing_combination.png:
SAme question, why keep the PNG instead of the SVG?

tblivet replied:
Good catch, that was an oversight.

jolelievre commented on admin-dev/themes/new-theme/img/empty_state/order.svg:
Same comment about the euro symbol

jolelievre commented on admin-dev/themes/new-theme/js/pages/product/combination/combination-modal/History.vue:
```diff
@@ -263,24 +263,24 @@
-    border-top-right-radius: 5px;
-    border-bottom-right-radius: 5px;
+    top: 0;
+    right: calc(-1 * var(--#{$cdk}size-32));
```
Are you sure the handle is still correctly aligned? It seems like you didn't change its size but its offset is way bigger

tblivet replied:
Yes, it's good for me !

jolelievre commented on admin-dev/themes/new-theme/js/pages/product/combination/combination-modal/History.vue:
```diff
@@ -293,9 +293,9 @@
-      right: -2rem;
-      background-color: #fff;
+      border: none;
+      right: calc(-1 * var(--#{$cdk}size-32));
```
SAme question here, isn't the offset too large now?

tblivet replied:
No, it's good for me !

jolelievre commented on admin-dev/themes/new-theme/package-lock.json:
It's impossible to review the changes in this file, can you explain what as changed?

jolelievre commented on admin-dev/themes/new-theme/package.json:
```diff
@@ -37,7 +37,7 @@
     "photoswipe": "^4.1.3",
     "postcss": "^8.4.4",
-    "prestakit": "^1.2.6",
+    "prestakit": "tblivet/prestashop-ui-kit_fork#reskin_update",
```
Same as in the default theme, before we merge this PR the UI kit should be merged and a new version released, probably a `2.0.0` version since I assume it contains some breaking changes?

tblivet replied:
Yes, you are right. I spoke with @matks about this. I think we will wait for the UI Kit to be merged and released as v2.0.0, and then update dependencies before the merge.

jolelievre commented on admin-dev/themes/new-theme/scss/components/_help_box.scss:
```diff
@@ -1,25 +0,0 @@
-.help-box {
```
Does it mean this component was completely removed?
Or was this moved somewhere else? In the UI kit maybe?

tblivet replied:
Krystian asked a similar question, here's my answer:
The style of this .hint-box is like an alert with a thicker left border, which to me makes no sense because we are trying to harmonize the BO. Therefore, it has no impact for me if the hint box displays as an alert, because the Twig file bootstrap_4_layout.html.twig uses this markup: `<div class="alert alert-info">`,so it will render as an info alert. That's why I have removed this part.

jolelievre commented on admin-dev/themes/new-theme/scss/components/_multishop-modal.scss:
```diff
@@ -10,40 +10,36 @@ $component-name: multishop-modal;
   &-dialog {
     width: 100%;
-    max-width: 39.375rem;
+    max-width: 40rem;
```
Why not use a variable here when all other hard-coded values have been replaced?

tblivet replied:
This is a high specific value, the variable set goes up to a maximum `size-320` = `20rem`

jolelievre commented on admin-dev/themes/new-theme/scss/components/layout/_main_header.scss:
```diff
@@ -83,27 +87,30 @@ $header-text-color: #4e6167 !default;
     height: 100%;
-    background: url("img/logo.png") no-repeat center left;
+    //background: url("img/logo.png") no-repeat center left;
+    background: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAUYAAAAkCAYAAAD1nJMVAAAACXBIWXMAABYlAAAWJQFJUiTwAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAv0SURBVHgB3Z3xdds4EsY/3bv/zx0cUkGcCpapYL0VhKkgcQWRK7BTgZUK7FQgpoJ4KxBSQXwV4DDGcEPLlPCBBEkov/fwHMcgQA6BwWAwAFfOuS2G86jph08PklarlU24Hr7+W//DYBij6+/cxxg5MHz293afcoG/p8r/+MOnc01nmgR5bqvpq08N++xa7ifMx6O/t79imfx9yT1V4LC+zPfIjL8Hka/I+k+9F/nddOvVJO3t ...
```
Why a hard-coded base64 image here instead of a file?

tblivet replied:
This was done a long time ago with the intent of simply editing the CSS file, but I've gone a bit further since then by modifying a few template, images... So, I'm going to refactor this to use a real SVG instead of base64.

jolelievre commented on admin-dev/themes/new-theme/scss/components/layout/_nav_bar.scss:
```diff
@@ -49,70 +58,40 @@
+        width: var(--#{$cdk}size-128);
+        height: var(--#{$cdk}size-24);
+        //background: url("#{$img-path}/ps-logo-white.svg") no-repeat;
+        background: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAUYAAAAkCAMAAADCQmMnAAAAAXNSR0IB2cksfwAAAAlwSFlzAAAWJQAAFiUBSVIk8AAAAEVQTFRFHR0bHR0bHR0bHR0bHR0bHR0bAAAAHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0bHR0buFvkHQAAABd0Uk5Tv6+fgFAgAEB/3+//z2AQj3BvXzCQoLC2y4BfAAAGb0lEQVR4nLWa6aKrKAyA3TngUu/oue//qFO1ZCNBOu3kXwGzfOyhVcWlbtquH4g0N/XDUBnS0Ubup/ZhHMNUz+RzZ317WIJWP1jYDJos ...
```
Same question, why not use an image that would be more maintainable and would have better performance (can be cached by the browser and zipped by the server)

tblivet replied:
This was done a long time ago with the intent of simply editing the CSS file, but I've gone a bit further since then by modifying a few template, images... So, I'm going to refactor this to use a real SVG instead of base64.

jolelievre commented on admin-dev/themes/new-theme/scss/img/helper-card/attributes@3x.svg:
Same comment about the euro symbol

jolelievre commented on admin-dev/themes/new-theme/scss/img/helper-card/features@3x.svg:
Same comment about the euro symbol

jolelievre commented on admin-dev/themes/new-theme/scss/pages/_cms_page.scss:
Why not remove the file completely if it's now useless?

tblivet replied:
Ok, I will delete it.

jolelievre commented on admin-dev/themes/new-theme/scss/pages/product/product_page.scss:
```diff
@@ -2465,6 +2443,11 @@ $product-page-padding-bottom: 80px !default;
     }
 
+    h2 {
+      padding-top: 0.5rem;
```
Why not use `var(--#{$cdk}size-8)` here?

jolelievre commented on admin-dev/themes/new-theme/static/helper-card/team@3x.png:
SAme question, is the PNG needed?

jolelievre commented on img/app_icon.png:
😢

jolelievre commented on src/PrestaShopBundle/Resources/views/Admin/Component/Layout/multistore_header.html.twig:
```diff
@@ -75,7 +75,7 @@
                     <i class="material-icons">check</i>
                   {% else %}
-                    <span class="multishop-modal-color" style="background-color:#25b9d7;"></span>
+                    <span class="multishop-modal-color" style="background-color:#a4dbe8;"></span>
```
Should this be moved into CSS?

jolelievre commented on src/PrestaShopBundle/Resources/views/Admin/Multistore/header.html.twig:
```diff
@@ -75,7 +75,7 @@
                     <i class="material-icons">check</i>
                   {% else %}
-                    <span class="multishop-modal-color" style="background-color:#25b9d7;"></span>
+                    <span class="multishop-modal-color" style="background-color:#a4dbe8;"></span>
```
Same question, should this be in CSS?

jolelievre reviewed (commented):
Thanks @tblivet 

Wow that as a long one 😅
I'm available if you want to discuss the comments live instead of via asynchronous comments

Hlavtox commented:
@tblivet Hi, thank you, I will check the PR tommorow or during the weekend again properly.

If there is something not clear or ruining the UX, I will put it to the review.

tblivet committed:
refactor: rebrand review

tblivet committed:
refactor: empty states + font display

tblivet committed:
refactor: hint-box alert default display

Hlavtox reviewed (dismissed):
Hi @tblivet, I did a first test. Looks million times better than the "edition". Can you please look at this and consult with a product team?

### Czech (latin-ext) fonts not working
![Snímek obrazovky 2024-07-31 111115](https://github.com/user-attachments/assets/4eff7f09-c72e-4e6f-a9f4-67ae7a2814d9)

### Left menu jumping
https://github.com/user-attachments/assets/02c65520-df73-4ea1-a9c1-e37a342768e7

### Primary color
Can you consider choosing any other color than black as primary? The primary button needs to be visible, so the user knows where to go, what action he can do. For example, choosing this purple color from the branding would make it much better.

![Snímek obrazovky 2024-07-31 111533](https://github.com/user-attachments/assets/2fb1f001-ef50-4e6b-b7ba-fc6b3b48c9b8)

### Menu styling
The menu being white as the rest of the page makes everything confusing. Can you consider making it dark grey or any other color? Now it looks like the rest of the page and the "barrier" between the content and menu is lost.

### Order page
Can you check this button, along with the dropdown? It's using some btn-action weird custom style and it's not styled. it would be awesome if you could have a look on that whole section, because what is there originally is terrible. :D

![Snímek obrazovky 2024-07-31 111802](https://github.com/user-attachments/assets/3b1e4ede-587f-4872-92a9-06d073e6d9f9)

### Mobile display of headers
Can you consider reducing the spacing and font-sizing on some elements on smaller displays, or making it somehow relative, using wv units? Most of the things doesn't fit.

Also, what about adding a thin border for the card header? You kept it for the footer, but maybe it would be good also for the header, to distinguish what is the card body and what is the header.

![Snímek obrazovky 2024-07-31 112047](https://github.com/user-attachments/assets/37b28f8e-de4d-4650-b2ab-345a54bee469)

tblivet commented:
Hi @Hlavtox, thank you for your feedback. I will take a look at it 👍

tblivet committed:
refactor: review + update package + lint

jolelievre commented on admin-dev/themes/default/scss/font.scss:
```diff
@@ -1,35 +1,102 @@
+$FontPathIBMPlexSans: "~@fontsource/ibm-plex-sans";
+$local-font: true;
+
+@mixin setFont() {
```
Is it possible to use the same principle you changed recently as in `_import.scss` in the UI kit

Or maybe import this file from UI kit directly? Or maybe it's not even needed anymore to do this here since the fonts are already included in the UI kit?

tblivet replied:
Yes, I have refactored it the same way 👍

Hlavtox commented:
@tblivet I am sure there will be no issue with that, as those are just basic UX rules. 👍

Even now our employees and my other clients complain and were very unhappy with the 1.7.7 order page and 8.1 product page layouts. All black design with washed out colors would kill it even more.

User must orientate on the page quickly. With current menu in different color, it's much easier, because you can immediately target the content part of the website, not spend time looking where the start of the content is. [Elements tend to be perceived into groups if they are sharing an area with a clearly defined boundary.
](https://lawsofux.com/law-of-common-region/) The same goes for button color. You come to a page - what should I do? Where should the next action go to? Cards are greyscale, tables are greyscale, links are black, buttons are black. Very difficult.

Wordpress admin (or the current design) is the perfect example of things that work. [Users spend most of their time on other sites. This means that users prefer your site to work the same way as all the other sites they already know.](https://lawsofux.com/jakobs-law/) There is a reason that Microsoft's Metro design with all blue and white was abandoned in favor of colorful icons.

----------

Lastly, I know that the project must adhere to the branding the company has paid for, but I think that for 99 % of future and current users of this software, a better selling point would be if the backoffice was usable on phones or if the software would calculate prices correctly. :-)

tblivet committed:
refactor: replace class containing ps version

jolelievre commented on admin-dev/themes/new-theme/.webpack/common.js:
```diff
@@ -387,7 +387,7 @@ module.exports = {
         type: 'asset/resource',
         generator: {
-          filename: '[hash].[ext]',
+          filename: '[hash][ext]',
```
Why do you need to change this? Doesn't it mean the generated files will look like `sqdfsdfwoff` instead of `sqdfsdf.woff` ?

tblivet replied:
I noticed that the resource was loaded like this: `6fbc1002a60aab967a20..woff2` with two dots, so I think the browser was friendly with its use.

jolelievre replied:
Ah ok, so I guess the `[ext]` includes the dot already

jolelievre committed:
Fix package-lock.json

Hlavtox commented on admin-dev/themes/new-theme/scss/components/layout/_nav_bar.scss:
```diff
@@ -426,7 +426,8 @@
     }
 
-    @media (max-height: 870px) {			// this two tabs need to be flipped, otherwise
+    @media (max-height: 1040px) {
```
This smells bad. :D No idea what magic is this, but targetting a specific pages in global component is not a good idea.

```css
#subtab-ShopParameters.ul-open,
#subtab-AdminAdvancedParameters.ul-open {
```

Also, what is `@media (max-height:` used for? It's very non-standard, I have never ever seen it used in real life. 🤔

jolelievre replied:
It's not the page, it targets the elements in the menu that are at the end I totally agree that this is quite dirty, if the structure of the menu is changed is not respected anymore then it will stop working But the purpose is not to refacto this in this PR

The new menu style is a bit wider/higher so the threshold for this rule also needs to be increased basically When the page has enough height there's no need for this because there's enough room to display the submenu

But I agree it would deserve a more generic and stable solution

tblivet replied:
@jolelievre has very well summed up what happen here, and it's clearly quite dirty, but it's not the purpose of the PR, even if we all agree that a better solution would be deserved.

jolelievre committed:
Update UI tests and fix a few bugs for new reskin

jolelievre reviewed (dismissed):
Thanks @tblivet

tblivet commented:
Hi @Hlavtox,
After discussing with the product/design team, I have some information to share with you.

**Primary color:** The primary color will remain dark with small touches of blue, like in the navbar.

**Menu styling:** The menu will stay white. Designers are not against working on a version with a darker background, but if needed, they will have to work on it. So, this might be addressed through an issue after the rebrand merge.

kpodemski commented:
We also need approval from @PrestaShop/ui-tests-maintainers

kpodemski commented:
@Hlavtox let's get it into QA, it will be easier to test for improvements after the merge :)

Hlavtox commented:
@tblivet OK, I will change it in another PR after discussion with product council. 👍

florine2623 reviewed (commented):
Hello @tblivet ,

Thanks for the PR !

I have some feedback :)

In Catalog > Products > Edit any product > Go to Details tab > See Feature block :
<img width="781" alt="Screenshot 2024-08-08 at 14 29 16" src="https://github.com/user-attachments/assets/83e250c1-6399-4627-9c98-2d76592f7ed5">

In Catalog > Products > Edit any product > Go to Stocks tab > Edit quantity (+ and -) > See Recent stock movements block :
I think the color for + and - stock movement label should be more differentiable, the black and grey colors look too similar.
<img width="622" alt="Screenshot 2024-08-08 at 14 37 16" src="https://github.com/user-attachments/assets/fad5f212-ceb8-4106-9401-5eefebc8c145">

Same thing in Catalog > Stocks > Movements tab :
<img width="1224" alt="Screenshot 2024-08-08 at 14 51 08" src="https://github.com/user-attachments/assets/15bcef41-08cd-4f0a-b988-67846b2f0371">

In Customer Service > Merchandise returns > The Status is a little cropped :
<img width="412" alt="Screenshot 2024-08-08 at 15 06 15" src="https://github.com/user-attachments/assets/11240268-6ab1-46d9-80f4-f8778a939918">

In Design > Theme & Logo > Pages Configuration Tab :
The arrow button are broken. 
Also I think the color palette is not correct, it shouldn't be blue.
<img width="1433" alt="Screenshot 2024-08-08 at 15 09 56" src="https://github.com/user-attachments/assets/6387a604-8a0f-46ac-8f1f-385d37e9c5cc">

In Design > Theme & Logo > Advanced Customization Tab :
The buttons shouldn't be blue, also the shape/size feel a little off. I couldn't tell you why 😅 
<img width="1104" alt="Screenshot 2024-08-08 at 15 52 45" src="https://github.com/user-attachments/assets/f7806e38-b621-4f79-b0c4-1e682dc83888">


Thank you !

tblivet committed:
fix: QA review

tblivet commented:
Hello @florine2623,

Thank you for the review 👍

Everything is now fixed, except for the part related to **Design > Theme & Logo > Pages Configuration** and **Design > Theme & Logo > Advanced Customization**. This is outside of the scope because a module handles this part. I just fixed the issue with the button arrow.

kpodemski commented on admin-dev/themes/default/scss/controllers/_themes.scss:
```diff
@@ -58,6 +58,21 @@
   padding-top: 0;
 }
 
+#psthemecusto {
```
@tblivet I'll approve but this should be moved to the module's code some other day

tblivet committed:
fix: QA review

florine2623 reviewed (approved):
Hello @tblivet ,

Thanks for the quick update ^^

All the bugs mentioned are fixed now 💪 

No more cropped feature detail :

<img width="826" alt="Screenshot 2024-08-09 at 16 03 14" src="https://github.com/user-attachments/assets/a574ed51-dd08-452e-b631-1b157a00b505">

New colors are implemented when adding ou removing stock on Product page and on Movement page :
<img width="613" alt="Screenshot 2024-08-09 at 16 02 43" src="https://github.com/user-attachments/assets/4f6f6c79-6c75-4221-b806-165cb4a4f70d">
<img width="1218" alt="Screenshot 2024-08-09 at 16 04 17" src="https://github.com/user-attachments/assets/4f8a4e3b-5667-44e8-8ea0-1c850f39ad66">

No more cropped options Status in RMA :
<img width="372" alt="Screenshot 2024-08-09 at 16 06 28" src="https://github.com/user-attachments/assets/75183d97-1e59-4fed-a127-6ed598aec016">

Arrow buttons are displayed correctly :
<img width="1182" alt="Screenshot 2024-08-09 at 16 13 32" src="https://github.com/user-attachments/assets/09ede799-3b1e-47a9-b7c0-5ded11467a00">

Last point verified with @tblivet.
Before:
<img width="1199" alt="Screenshot 2024-08-09 at 16 28 08" src="https://github.com/user-attachments/assets/fc900979-9f3a-4971-8954-f067153996b5">
After:
<img width="940" alt="Screenshot 2024-08-09 at 18 13 19" src="https://github.com/user-attachments/assets/42081f81-7569-45e2-b1a3-ff30d4381fda">

Noted for the last point, I understand. I just wanted to let you know in case you had some directives about that ^^

I have checked the whole PR entirely once again, it looks good to me !
It is QA ✅ 

Thanks!

jolelievre commented:
Thanks @tblivet and @florine2623
